lib/Dialect/OM/Transforms/LinkModules.cpp

```diff
@@ -57,14 +59,25 @@ LogicalResult ModuleInfo::initialize() {
   for (auto &op : llvm::make_early_inc_range(module.getOps())) {
     if (auto classLike = dyn_cast<ClassLike>(op))
       symbolToClasses.insert({classLike.getSymNameAttr(), classLike});
-    else
+    else if (auto hwMod = dyn_cast<HWModuleLike>(op))
+      symbolToHWModules.insert({hwMod.getModuleNameAttr(), hwMod});
+    else {
       op.erase();
```

Comment:
I guess we need to preserve/resolve operations with SymbolOpInterface. Linking OM was easy because other than symbols on classes were not used but if we need to link HW, we basically need to handle all SymbolOpInterface i guess.

Reply:
Updated the implementation to support any ops declaring symbols.